cpp/src/cylon/table.cpp

```diff
@@ -1056,26 +1126,124 @@ Status Equals(const std::shared_ptr<cylon::Table>& a, const std::shared_ptr<cylo
   return Status::OK();
 }
 
+static Status RepartitionToMatchOtherTable(const std::shared_ptr<cylon::Table> &a, const std::shared_ptr<cylon::Table> &b, std::shared_ptr<cylon::Table> * b_out) {
+  int world_size = a->GetContext()->GetWorldSize();
+  int num_row = a->Rows();
+
+  if(num_row == 0) {
+    return Status::OK();
+  }
```

Comment:
I think since we are returning `Status::OK`, we need to set `b_out`. We can simply assign `a` to it.